strawberryfields/engine.py

```diff
@@ -182,58 +285,34 @@ def _run_program_locally(self, prog, **kwargs):
                 try:
                     temp = cmd.op.decompose(cmd.reg)
                     # run the decomposition
-                    applied_cmds = self._run_program_locally(temp, **kwargs)
+                    applied_cmds = self._run_program(temp, **kwargs)
                     applied.extend(applied_cmds)
                 except NotImplementedError as err:
                     # simplify the error message by suppressing the previous exception
                     raise err from None
         return applied
 
-    def run(self, program, return_state=True, modes=None, compile=True, **kwargs):
-        """Execute the given program by sending it to the backend.
+    def run(self, program, *, compile=True, modes=None, state_options={}, **kwargs):
```

Comment:
If you have a `state_options` argument, then is there any more need for `kwargs`?

Reply:
Yep, curious about this too

Reply:
`kwargs` get passed on to `Operation.apply()`, which uses `eval_params` if it appears, and then on to the `backend.measure_*` methods that look for various other options in them. It seems these include at least feed dictionaries for TF, homodyne sampling options for the Fock backend and quadrature eigenstate approximation parameters for the Gaussian backend.